src/kani-compiler/rustc_codegen_kani/src/codegen/statement.rs

```diff
@@ -78,18 +79,24 @@ impl<'tcx> GotocCtx<'tcx> {
                     if *expected { r } else { Expr::not(r) }
                 };
 
-                Stmt::block(
-                    vec![
-                        cond.cast_to(Type::bool()).if_then_else(
-                            Stmt::goto(self.current_fn().find_label(target), loc.clone()),
-                            None,
-                            loc.clone(),
-                        ),
-                        Stmt::assert_false(&format!("{:?}", msg), loc.clone()),
-                        Stmt::goto(self.current_fn().find_label(target), loc.clone()),
-                    ],
-                    loc,
-                )
+                let mut msg_str = format!("{:?}", msg);
```

Comment:
What if we tagged EVERY assertion with a check id?

Reply:
Currently, this would be a bit problematic because the `expected` test suite mostly uses the old format for which the CBMC parser is not run. And it looks like CBMC currently doesn't have a way to tag assertions (see https://github.com/diffblue/cbmc/issues/6692).